lib/elixir_search_extractor_web/views/api/v1/error_view.ex

```diff
@@ -0,0 +1,7 @@
+defmodule ElixirSearchExtractorWeb.Api.V1.ErrorView do
+  use JSONAPI.View, type: "error"
+
+  def render("error.json", %{errors: detail}) do
```

Comment:
How about we provide something like this 🤔 

```elixir
def render("error.json", %Ecto.Changeset{} = changeset) do
 %{errors: [%{detail: EctoErrorHelper.translate_error(changeset)}]}
end

def render("error.json", message) do
  %{errors: [%{detail: reason}]}
end
```

So we can simplify when rendering the error from controller

```elixir
|> put_view(ErrorView)
|> render("error.json", changeset)

|> put_view(ErrorView)
|> render("error.json", reason)
```

Not sure if there is any restriction when using `JSONAPI.View`

Reply:
I could do something close. Just need to send some key value pair for `JSONAPI` to work with data. Added in [1dbb532](https://github.com/rafayet-monon/elixir_search_extractor/pull/53/commits/1dbb53226c9e26174e5dbcb4a684df5fd02418b7)